components/FormForConditions.js

```diff
@@ -0,0 +1,271 @@
+import { useForm } from 'react-hook-form';
+import useStore from '../hooks/useStore';
+import useHydration from '../hooks/useHydration';
+import styled from 'styled-components';
+import Link from 'next/link';
+import Toast from './Toast';
+import NavWrapper from './NavWrapper';
+import BackIcon from '../public/icon-left.svg';
+import { BigButton, SmallButton, ContentWrapper } from './Buttons';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { SchemaConditions } from './FormValidation';
+import { GetCleanNumber } from '../hooks/useCalculation';
+import React, { useState, useRef, useEffect } from 'react';
+import { Box, LinearProgress } from '@mui/material';
+import Modal from './Modal';
+import { StyledInput, StyledHeadline } from './FormStyledComponents';
+
+export default function FormForConditions() {
+  const hydrated = useHydration();
+  const [open, setOpen] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const timer = useRef();
```

Comment:
Timer sind schwer zu überblicken, aber wenn ich das hier richtig sehe, solltest du den timer und die beiden Funktionen die ihn verwenden, in den useEffect können dann brauchst du auch keinen useRef. Das du den Timer dortabräumst ist super!

Reply:
das habe ich ehrlich gesagt nur "übernommen" und verstehe nicht 100% was es macht :)